pkg/controllers/rollout/controller.go

```diff
@@ -333,25 +331,30 @@ func (r *Reconciler) pickBindingsToRoll(ctx context.Context, allBindings []*flee
 	readyTimeCutOff := time.Now().Add(-time.Duration(*crp.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds) * time.Second)
 
 	// classify the bindings into different categories
-	// TODO: calculate the time we need to wait for the first applied but not ready binding to be ready.
+	// Wait for the first applied but not ready binding to be ready.
 	// return wait time longer if the rollout is stuck on failed apply/available bindings
+	minWaitTime := time.Duration(*crp.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds) * time.Second
+	allReady := true
 	crpKObj := klog.KObj(crp)
-	for idx := range allBindings {
-		binding := allBindings[idx]
+	for _, binding := range allBindings {
```

Comment:
do you know the difference between this and the original?

Reply:
Yes, did not mean to change it. It was part of a different refactoring approach. Restored to the original.